src/text/raw_text_reader.rs

```diff
@@ -91,22 +105,25 @@ impl<T: TextIonDataSource> RawTextReader<T> {
             }
 
             // Otherwise, see if the next token in the stream is an Ion Version Marker.
-            match self.parse_next(ion_version_marker) {
-                Ok(Some((1, 0))) => {
+            match self.parse_next_nom(ion_version_marker) {
+                RootParseResult::Ok((1, 0)) => {
                     // We found an IVM; we currently only support Ion 1.0.
                     self.current_ivm = Some((1, 0));
                     return Ok(());
                 }
-                Ok(Some((major, minor))) => {
+                RootParseResult::Ok((major, minor)) => {
                     return decoding_error(format!(
                         "Unsupported Ion version: v{}.{}. Only 1.0 is supported.",
                         major, minor
                     ));
                 }
-                // I/O errors are fatal; we cannot keep parsing.
-                Err(e @ IonError::IoError { .. }) => return Err(e),
+                RootParseResult::Failure(error_message) => {
+                    return decoding_error(error_message);
+                }
```

Comment:
🗺️ This entire PR was effectively to allow this change: when the reader is checking to see if the next item in the stream is an IVM, it can distinguish between a parsing error ("text did not match") and a fatal error (IO error, invalid utf8, invalid value, etc) and report the latter to the user.